beacon_node/beacon_chain/src/data_availability_checker/state_lru_cache.rs

```diff
@@ -26,7 +27,7 @@ pub struct DietAvailabilityPendingExecutedBlock<E: EthSpec> {
     parent_block: SignedBeaconBlock<E, BlindedPayload<E>>,
     parent_eth1_finalization_data: Eth1FinalizationData,
     confirmed_state_roots: Vec<Hash256>,
-    consensus_context: ConsensusContext<E>,
+    consensus_context: OnDiskConsensusContext,
```

Comment:
`DietAvailabilityPendingExecutedBlock` is constructued on every block, and since `OnDiskConsensusContext` drops `indexed_attestations`, that cache would be dropped every block.

Could fix this easily by keeping the `indexed_attestations` field in `OnDiskConsensusContext` with `#[ssz(skip_serializing, skip_deserializing)]`.

Reply:
Great catch, just pushed a fix